### ISSUE TITLE
[BUG] call ended callback is triggered but actually call connection/stream is not closed, callee and caller can still talk to each other.

### ISSUE THREAD

**Issue by chenghung** [twilio/twilio-voice.js#29]
- [x] I have verified that the issue occurs with the latest twilio.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-voice.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-voice.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/TwilioDevEd/voice-javascript-sdk-quickstart-node) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

this issue is hard to reproduce and rarely happneded.

```js
    device.on(Device.EventName.Incoming, (call: any) => {
      if (callEndedCallback) {
        call.once('disconnect', () => {
          console.log('call is disconnected');
          callEndedCallback();
        });
      }      
```

**Expected behavior:**

while callEndedCallback is triggered, this call and its stream should be closed and user should not able to talk or hear any voice from caller. 

**Actual behavior:**

call conversation still keep alive after received "disconnect" event.

**Software versions:**

 - [x] Browser(s): Firefox, Google Chrome
 - [x] Operating System: Linux / Windows 10
 - [x] twilio.js: v2.0.1
 - [x] Third-party libraries (e.g., Angular, React, etc.): react

**Comment by ryan-rowland** [comment]
Hi @chenghung, thanks for bringing this to our attention. This isn't something we've seen in our testing, and as it is hard to reproduce I haven't been able to reproduce this on my end locally. Please re-open if you're able to provide some more information that might lead to a reproduction of the issue so we can investigate.

**Comment by alfietinkerve** [comment]
I am facing this exact same issue in twilio v2.1.1. Only that it happens everytime. 

Specifically on these use cases:-

**Outbound**

Start a call -> ringing -> caller disconnect() -> receiver still ringing
Start a call -> ringing -> receiver reject -> caller still ringing

**Inbound**

Incoming call -> ringing -> receiver reject -> caller still ringing
Incoming call -> ringing -> accepted -> disconnect() -> caller still in call

The calls eventually drops but only after a long timeout (close to 20 seconds) or the other party drops as well.
For inbound, I keep receiving incoming calls as the caller is still ringing so I had to unregister to prevent which is not a good practice.
I just call the SDK function disconnect/reject to drop the call from the front end.

**Comment by mhuynh5757** [comment]
Hi, thanks for bringing this issue up to our attention. Could we ask for more details with regards to this?

Did you encounter after upgrading from an older version of the SDK? If so, which version?

Are these client to client calls, or PSTN calls?

Would you mind sharing some `callSid`s with us?

We're not able to reproduce this within our environments using client to client calls.

**Comment by karan-bacancy** [comment]
@alfietinkerve @mhuynh5757  what's workaround you had done for this issue ? can you please let me know ?

**Comment by charliesantos** [comment]
Hi @karan-bacancy , we have not been able to reproduce this issue. Can you please provide information @mhuynh5757 asked? As we're not able to reproduce using @alfietinkerve 's repro steps, do you have a more detailed reproduction steps?

**Comment by charliesantos** [comment]
Closing due to non-reproducibility and inactivity. Please feel free to follow up and provide the requested information above so we can investigate further.
